session/store.js

```diff
@@ -47,12 +47,27 @@ util.inherits(Store, EventEmitter)
  * @api public
  */
 
-Store.prototype.regenerate = function(req, fn){
+Store.prototype.regenerate = function(req, options, fn){
   var self = this;
-  this.destroy(req.sessionID, function(err){
+  if (options instanceof Function) {
+    fn = options;
+    options = {};
+  }
+  options = options || {};
```

Comment:
Back in commit https://github.com/expressjs/session/commit/d95275bd5345b161585ec7c6de22bcd77b484a65 we tried to fix some deopts by avoiding argument reassignments. Please avoid argument reassignments in this code as well :)